workers/generic-worker/mounts.go

```diff
@@ -676,6 +676,8 @@ func extract(fsContent FSContent, format string, dir string, task *TaskRun) erro
 		return archiver.Rar.Open(cacheFile, dir)
 	case "tar.bz2":
 		return archiver.TarBz2.Open(cacheFile, dir)
+	case "tar.xz":
```

Comment:
I could not find where in the code other cases where tested, but I'm happy to add tests if pointed to.